src/components/VocabularyDrill/VocabularyDrill.tsx

```diff
@@ -267,8 +268,15 @@ export const VocabularyDrill: FC<VocabularyDrillProps> = ({
     setDisableTools(false);
     setDisableNextButton(true);
 
-    setScore(score - (activeContentType.current?.getScore() ?? 0));
-    setMaxScore(maxScore - (activeContentType.current?.getMaxScore() ?? 0));
+    if (page > 0) {
+      setScore(score - (activeContentType.current?.getScore() ?? 0));
+      setMaxScore(maxScore - (activeContentType.current?.getMaxScore() ?? 0));
+    }
+    else {
+      // Reset the score
+      setScore(0);
+      setMaxScore(0);
```

Comment:
Just an informational note, nothing to do: Both H5P and xAPI allow the maximum score to be 0. Unfortunately moodle HQ did not seem to have a proper understanding of either and built a flaw into their H5P integration. Completion detection for H5P activities (which should not be bound to scores) does not work if the maximum score is set to 0 and reported via xAPI. This is not an issue here - the maximum score should only be 0 if there's no exercise at all, rendering the Vocabulary Drill task useless anyway - but maybe this info is relevant in the future, in particular for content types that can be _completed_ but not necessarily scored.

Reply:
Thanks for the info. Noticed that my last commit was not the fix for the problem I found